linalg/solvers.hpp

```diff
@@ -181,6 +182,8 @@ class OperatorChebyshevSmoother : public Solver
 
    void Mult(const Vector&x, Vector &y) const;
 
+   void MultTranspose(const Vector &x, Vector &y) const { Mult(x, y); }
```

Comment:
I am not sure if the Chebyshev smoother is symmetric as it is implemented now. A Chebyshev polynomial smoother requires, in general, a matrix which is scaled such that the entries on the diagonal are one. Here, we work with `D^(-1) * A` which is non-symmetric, but in theory we would rather want to use `D^(-1/2) * A * D^(-1/2)`.

In practice, this doesn't seem to be an issue. Do you maybe know more about this?

Reply:
Looking at the code in `OperatorChebyshevSmoother::Mult`, I get
```
y_0 = c_0 D^{-1} b
y_1 = (c_0 D^{-1} + c_1 D^{-1} A D^{-1}) b
y_2 = (c_0 D^{-1} + c_1 D^{-1} A D^{-1} + c_2 D^{-1} A D^{-1} A D^{-1}) b
```
These all look symmetric to me, but I might be missing something?

Reply:
You are correct. I was thinking too complicated and made a mistake. Your changes look good to me.